### ISSUE TITLE
Export `defaultSlot` in RAC or a way to workaround required slot errors

### ISSUE THREAD

**Issue by jrmyio** [adobe/react-spectrum#5671]
### Provide a general summary of the feature here

We are using the `Provider` from react-aria-components to create our own components. 

In order to allow components with and without slots inside a `Provider` we want to be able to configure the `default` so that we do not run into the error `A slot prop is required`.

However, since the `default` is a symbol and not exported in the `react-aria-components` we cannot use this pattern.

See:
https://github.com/adobe/react-spectrum/blob/ebc610d3caaf59461bd03b5a55c30be0f318c29e/packages/react-aria-components/src/utils.tsx#L27
 https://github.com/adobe/react-spectrum/blob/ebc610d3caaf59461bd03b5a55c30be0f318c29e/packages/react-aria-components/src/index.ts#L18

### 🤔 Expected Behavior?

Be able to use `defaultSlot` in custom components so we can prevent the error.

### 😯 Current Behavior

You run into an error `A slot prop is required` when not specifying a slot for a component while having siblings with slots.

### 💁 Possible Solution

When building our design system on top of RAC i felt that the slot errors are a quite the hassle. We've seen the errors come up multiple times and sometimes this was good and it catched a mistake on our part, but in general I think using a default empty object for the default instead of erroring out is a better default.

The other option is to add `defaultSlot` to react-aria-components exports to allow users to use the symbol just how RAC components uses it internally for its components.

### 🔦 Context

Sometimes we have buttons in our UI that have slots and are configured by a `Provider` higher up in the tree. Sometimes we have buttons that don't use context and thus don't have a slot.

We want to support both.

### 💻 Examples

[_Codesandbox example_](https://codesandbox.io/p/sandbox/rac-provide-nested-slots-n44xk2?file=%2Fsrc%2FApp.tsx%3A45%2C18)

**Comment by LFDanLu** [comment]
I'm a bit unsure as to the exact parts being used here, just to make sure I understand: you are creating your own components and are using our RAC Provider within those components to shuttle props to slots like our components?

**Comment by jrmyio** [comment]
> I'm a bit unsure as to the exact parts being used here, just to make sure I understand: you are creating your own components and are using our RAC Provider within those components to shuttle props to slots like our components? Maybe a small example in a codesandbox would be helpful in clearing this up.

Yes this is exactly what we are doing. Here is a codesandbox (also added this to OP):

https://codesandbox.io/p/sandbox/rac-provide-nested-slots-n44xk2?file=%2Fsrc%2FApp.tsx%3A45%2C18

**Comment by LFDanLu** [comment]
Thanks for the sandbox. We are fine with exposing the `defaultSlot`, but we will also need to document its usage in the [Advanced Customization](https://react-spectrum.adobe.com/react-aria/advanced.html) part of the docs.

**Comment by sookmax** [comment]
Looks like the following line was added in https://github.com/adobe/react-spectrum/pull/5052, roughly 4 months ago.
https://github.com/adobe/react-spectrum/blob/46ae3314a2e2c1ab856c82c8d9dc941e8c70016e/packages/react-aria-components/src/utils.tsx#L167-L170

I created [a sandbox](https://codesandbox.io/p/devbox/hopeful-architecture-7mtp9w?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clrt7xyvw00073b6fmd6lfwij%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clrt7xyvw00023b6f2mtcej7l%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clrt7xyvw00043b6fk9l864vx%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clrt7xyvw00063b6fnimk2jbl%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clrt7xyvw00023b6f2mtcej7l%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrt7xyvw00013b6fc0ftebnr%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clrt7xyvw00023b6f2mtcej7l%2522%252C%2522activeTabId%2522%253A%2522clrt7xyvw00013b6fc0ftebnr%2522%257D%252C%2522clrt7xyvw00063b6fnimk2jbl%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrt7xyvw00053b6fbu6nkjm5%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clrt7xyvw00063b6fnimk2jbl%2522%252C%2522activeTabId%2522%253A%2522clrt7xyvw00053b6fbu6nkjm5%2522%257D%252C%2522clrt7xyvw00043b6fk9l864vx%2522%253A%257B%2522id%2522%253A%2522clrt7xyvw00043b6fk9l864vx%2522%252C%2522activeTabId%2522%253A%2522clrt7xyvw00033b6ftaxzd1ye%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrt7xyvw00033b6ftaxzd1ye%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clrt7yknf00633b6fzw5fdqwl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clrt7ykvq004ndpip2tto6257%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) using `"react-aria-components": "^1.0.1"`, and it seems to work if you specify _explicit_ `null` to slot prop like:
```tsx
<Button
  slot={null}
  onPress={() =>
    console.log("I am a custom button but I need a slot")
  }
>
  Some other button
</Button>
```

**Comment by nmanumr** [comment]
I need this badly. also, Can we have `slotCallback` exported too?

**Comment by LFDanLu** [comment]
I think that should be fine as well. Do you need it for your types?

**Comment by devongovett** [comment]
I was hoping to remove slotCallback actually. I don't think we use it anymore. What do you need it for?

**Comment by nmanumr** [comment]
Actually, I had a use cases where i wanted to known when the child element is rendered. slotCallback came handy there. I can share the example code if you want.

**Comment by devongovett** [comment]
Ah. Another way to do that is with a callback ref:

```js
let [hasSlot, setHasSlot] = useState(false);
let ref = useCallback(el => {
  setHasSlot(!!el);
}, []);

<SomeContext.Provider value={{ref}}>
  {children}
</SomeContext.Provider>
```
